pkgs/tools/bootloaders/refind/default.nix

```diff
@@ -34,6 +34,9 @@ stdenv.mkDerivation rec {
   patches = [
     # Removes hardcoded toolchain for aarch64, allowing successful aarch64 builds.
     ./0001-toolchain.patch
+    # Avoid leaking the build timestamp
+    # https://sourceforge.net/p/refind/code/merge-requests/53/
+    ./reproducible.patch
```

Comment:
```suggestion
    ./0002-preserve-dates.patch
```

Reply:
Ah, sorry, I had merged already but am happy to follow up.

Is the number prefix really useful? The ordering is already specified by the list of patches, having the ordering in the filenames as well just seems to make things messy when patches are added or removed later. On the other hand, indeed it seems somewhat common: about 500 of the 3700 patches in nixpkgs seem to be numbered that way.

Reply:
I think you just need to use git format-patch IIRC.

Reply:
I mean, I can, but I'm not sure I understand the benefit.

(indeed `preserve-dates` might have been a better name though)

Reply:
Yeah I don't really understand the benefits either.

Reply:
Right, so it makes it easier to work with `git am` and rebase all patches in one go. Good to know - though indeed probably not universally necessary.

Reply:
We should prefer `git format-patch` patches all the time. Not for the filename, but for the additional information in its header, and tracking authorship information.

Reply:
Good idea. Could y'all add those recommendations to https://github.com/NixOS/nixpkgs/tree/master/pkgs#patches ?

Reply:
> We should prefer `git format-patch` patches all the time. Not for the filename, but for the additional metadata in its header, and tracking authorship information.

I wouldn't want to write that as a hard requirement. It doesn't work when changes are done on top of a downloaded tarball and we already have most of the metadata like who wrote it, when it was written and there should be a short comment above the patch why we need it.

Reply:
My main concern with the numbering and the name is that patches are meant to be applied in sequence. (Sometimes patches can be applied unordered, but sometimes not.)
(Except maybe in Pijul, but I digress.)

Yes, lists in Nix are positional, but since sometimes the patch can be useful outside Nixpkgs - by a question of principle:  we pick patches from other distros, and they can pick patches from us - it is a good idea to put an order on the patches.

P.S.: git-format-patch can be used outside git? Sometimes the patch comes from other VCSes...